### PR TITLE
Add missing preview fields

### PR DESCRIPTION
**Pull request by tomaskikutis**
SDESK-5624

Preview fields like fields in authoring were hardcoded and it was hard to follow for which fields preview was implemented. I went over all hardcoded and custom fields and have built a mapper to field types which at the moment we have 9. I have decoupled the preview logic from field ids so all preview components now only rely on field type.

### CODE REVIEW

File: scripts/apps/authoring/preview/field-types/urls.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    urls: Array<{url: string; description: string}>;
+}
+
+export class UrlsPreview extends React.Component<IProps> {
+    render() {
+        return (
+            <div>
+                {
+                    this.props.urls.map(({url, description}) => (
+                        <div key={url}>{description}({url})</div>
```
Comment (pablopunk) [inline scripts/apps/authoring/preview/field-types/urls.tsx:13]: should urls be inside an `a` tag so the user can just click on it?

File: scripts/apps/authoring/preview/getAuthoringField.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+import {getRelatedArticles, getRelatedMedia} from '../authoring/controllers/AssociationController';
+
+export function getAuthoringField(
+    fieldId: string,
+    item: IArticle,
+    customFieldsVocabularies: Array<IVocabulary>,
+): IAuthoringField {
+    const customField = customFieldsVocabularies.find(({_id}) => _id === fieldId);
+
+    if (customField?.field_type === 'text') {
```
Comment (pablopunk) [inline scripts/apps/authoring/preview/getAuthoringField.ts:13]: This function looks more like a `switch` instead of so many `else if`

**Review by pablopunk — APPROVED** [review]

File: scripts/apps/authoring/preview/field-types/urls.tsx (hunk 1 above)
Comment (tomaskikutis) [reply scripts/apps/authoring/preview/field-types/urls.tsx]: It's meant to be printed so the link has to be inline. I have added a fix to make it clickable.

**Review by pablopunk — APPROVED** [review]
